src/generate-spec.js

```diff
@@ -22,6 +22,13 @@ var declarations = parse(specSource);
 
 var database = Object.create(null);
 
+var tailNames = {
```

Comment:
I'd like to see a comment about what this means and why we're doing it here instead of in the spec.